core/alsp_src/builtins/blt_shl.pro

```diff
@@ -59,15 +61,46 @@ start_shell0(DefaultShellCall)
 #endif
 */
 	(clause(dvf,_) -> qkc ; true),
-%	dvf,
 	load_cl_files(CLInfo),
 	process_cl_asserts(CLInfo),
 	!,
-		%% 
 	ss_load_dot_alspro(CLInfo),
+
+	check_setup_history_file,
+	check_load_prev_history,
+
 	setup_init_goal(CLInfo, ShellCall),
 	user:ShellCall.
 
+check_setup_history_file
+	:-
+	user:history_file_locn(HFL),
+	!,
+	setup_history_file(HFL).
+check_setup_history_file.
+
+setup_history_file(home)
+	:-!,
+	getenv('HOME', UserHomePath),
+	HistoryFile = '.alspro_history',
```

Comment:
The file name `.alspro_history` is repeated many times. It should be defined in one place (in case we ever need to adjust it).